src/devices/fineoffset.c

```diff
@@ -267,14 +267,17 @@ static int fineoffset_WH0530_callback(bitbuffer_t *bitbuffer) {
     }
 
     const uint8_t id = (buffer[0]<<4) | (buffer[1]>>4);
-    const float temperature = (float)((uint16_t)(buffer[1] & 0xF)<< 8 | buffer[2]) / 10.0 - 40.0;
-    const float rain = 0.3 * (((uint16_t)buffer[4] << 8) | buffer[3]);
+    const uint8_t battery_low = (buffer[1] & 0x8);
+    const float temperature = (float)((uint16_t)(buffer[1] & 0x7)<< 8 | buffer[2]) / 10.0 - 40.0;
```

Comment:
Hmm. I just had to read up on that (and test :-). You are right that it will be promoted to int and it works without the uint16_t cast. It would only be a problem on a 16 bit CPU, where it would be promoted to a 16 bit signed int and could be shifted into a negative number. However that is mostly a theoretical problem by now....
However, I am still a big believer in being explicit about the intent and not rely upon the more subtle compiler rules. I have been burned before by integer promotion. :-)